src/Microsoft.PowerShell.Commands.Utility/commands/utility/FormatAndOutput/format-hex/Format-Hex.cs

```diff
@@ -0,0 +1,395 @@
+            }
+        }
+
+        #endregion
+
+        #region Paths
+
+        /// <summary>
+        /// Validate each path provided and if valid, add to array of paths to process.
+        /// If path is a literal path it is added to the array to process; we cannot validate them until we
+        /// try to process file contents.
+        /// </summary>
+        /// <param name="path"></param>
+        /// <param name="literalPath"></param>
+        /// <returns></returns>
+        private List<string> ResolvePaths(string[] path, bool literalPath)
+        {
+            List<string> pathsToProcess = new List<string>();
+            ProviderInfo provider = null;
+
+            if (literalPath)
+            {
+                foreach (string currentPath in path)
+                {
+                    string newPath = Context.SessionState.Path.GetUnresolvedProviderPathFromPSPath(currentPath);
+                    pathsToProcess.Add(newPath);
+                }
+            }
+            else
+            {
+                foreach (string currentPath in path)
+                {
+                    try
+                    {
+                        Collection<string> newPaths = Context.SessionState.Path.GetResolvedProviderPathFromPSPath(currentPath, out provider);
+
+                        if (!provider.Name.Equals("FileSystem", StringComparison.OrdinalIgnoreCase))
+                        {
+                            // Write a non-terminating error message indicating that path specified is not supported.
+                            string errorMessage = StringUtil.Format(UtilityCommonStrings.FormatHexOnlySupportsFileSystemPaths, currentPath);
+                            ErrorRecord errorRecord = new ErrorRecord(new ArgumentException(errorMessage),
+                                "FormatHexOnlySupportsFileSystemPaths",
+                                ErrorCategory.InvalidArgument,
+                                currentPath);
+                            WriteError(errorRecord);
+                            continue;
+                        }
+
+                        if (newPaths.Count > 1)
+                        {
+                            // Write a non-terminating error message indicating that path specified is resolving to multiple file system paths.
+                            string errorMessage = StringUtil.Format(UtilityCommonStrings.FormatHexResolvePathError, currentPath);
+                            ErrorRecord errorRecord = new ErrorRecord(new InvalidDataException(errorMessage),
+                                "FormatHexResolvePathError",
+                                ErrorCategory.InvalidData,
+                                currentPath);
+                            WriteError(errorRecord);
+                        }
+                        else
+                        {
+                            pathsToProcess.Add(newPaths[0]);
+                        }
+                    }
```

Comment:
yep - looking below, it appears that it just loops through the provided paths, it seems reasonable to support more than one resolved path

---
In reply to: [105987086](https://github.com/PowerShell/PowerShell/pull/3320#discussion_r105987086) [](ancestors = 105987086)